Added tests, fixed global var dependence, fixed addInformation method

diego-d5000 opened the pull request:
This PR:
- Add some test following examples in ./examples.
- Modify resources (shipments and waypoints) classes to return a closure function, avoiding the dependence to a global var "iupick", this allow use of multiple instances of this iupick client lib to add more flexibility of use. (based on closed issue #1 'Module depends on global var')

Vanclief commented:
Hello Diego,

 I see the point of managing like this the modules, and adding tests was a very nice addition. When checking the PR, not all tests passed, I am adding the output of running the tests. As I said previously, I don't longer maintain this library (I checked the PR out of curiosity), so I do not feel I should merge this without @davicin-pola @nogalpi approval. But it looks good.

Best

**npm test**
```
2 failing

  1) Waybill
       Waybill process
         should pass through a waybill process successful:
     Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.


  2) Waybill
       track
         return a tracking code status:
     Uncaught AssertionError: expected { Object (shipper_address, recipient_address) } to be a string
```

diego-d5000 commented:
Yes, thank you for check it.

Yes, the first error, as I say in my previous comment, the waybill process on `iupick.shipment.addInformation` throws an error related to the type of postal_code field in address.

I debug this point and I found this error with the last commit with corrected postal_code type following the example as is, and I got this error:

``` json
{  
   "shipper_address":{  
      "postal_code":{  
         "non_field_errors":[  
            "Invalid data. Expected a dictionary, but got int."
         ]
      }
   },
   "recipient_address":{  
      "postal_code":{  
         "non_field_errors":[  
            "Invalid data. Expected a dictionary, but got str."
         ]
      }
   }
}
```
Taking the address example returned by the `getWaypointInformation` :

```json
{  
   "city":"México",
   "line_one":"Jose Maria Castorena #324-4",
   "line_two":"Jesus del Monte y Antonio Ancona",
   "postal_code":{  
      "state":{  
         "name":"Distrito Federal",
         "code":"DF"
      },
      "code":"05200",
      "latitude":"19.3699724849000000000000",
      "longitude":"-99.2888588093000000000000"
   }
}
```

postal_code must receive and object/dictionary/hashmap and the prop/key "code" must contain the true postal_code. I tried this by doing...

```js
const shipperAddress = iupickClient.createAddress(
        city = 'Querétaro',
        lineOne = 'Epigmenio Gonzáles 500',
        postalCode = { code: 76130 },
        lineTwo = '',
        neighborhood = 'Momma'
      );
```
But I got the next errror:

```json
{  
   "shipper_address":{  
      "postal_code":{  
         "state":[  
            "This field is required."
         ]
      }
   },
   "recipient_address":{  
      "postal_code":{  
         "non_field_errors":[  
            "Invalid data. Expected a dictionary, but got str."
         ]
      }
   }
}
```

I think that this error in the tests denotes and error in correct data structures synchronisation between API and client. I know that is not more your responsibility, so I will communicating with @nogalpi and @davicin-pola  but thank you for check my PR and run the tests :)

BTW, the second error is due to mocha timeout, it can be fixed by passing arguments --timeout in mocha opts.

diego-d5000 commented:
@davicin-pola,

I think that this PR is ready, finally all tests pass successful now. I added the following changes:

- Modified tests to show what data to send to the API, instead of send something like this:

```js
var shipperAddress = iupick.createAddress(
  city = 'Querétaro',
  lineOne = 'Epigmenio Gonzáles 500',
  postalCode = 76130,
  lineTwo = '',
  neighborhood = 'Momma'
);
```
it must be send an object in postalCode property/key

```js
const shipperAddress = iupickClient.createAddress(
        city = 'Querétaro',
        lineOne = 'Epigmenio Gonzáles 500',
        postalCode = {
          state:
            {
              name: "Querétaro",
              code: "QT"
            },
          code: "05200",
        },
        lineTwo = '',
        neighborhood = 'Momma'
      );
```

- Modified `addInformation` method to fill recipient address required by the API, fetching the waypoint address from waypoint id.
